File: TikTok/app/src/main/java/com/example/tiktok/MainActivity.java
package com.example.tiktok;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText etUserName, etPassword;
    private Button btnLogin;
    private CheckBox remember;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        etUserName = findViewById(R.id.EtUserName);
        etPassword = findViewById(R.id.EtPassword);
        remember = findViewById(R.id.checkbox1);
        btnLogin = findViewById(R.id.btnLogin);

        SharedPreferences sharedPreferences = getSharedPreferences("checkbox", MODE_PRIVATE);
        String checkbox = sharedPreferences.getString("remember","");
        if(checkbox.equals("true")){

            Intent intent = new Intent(MainActivity.this, TikTokActivity.class);
            startActivity(intent);
        }else if(checkbox.equals("false")){
            Toast.makeText(MainActivity.this,"plaease sign in",Toast.LENGTH_SHORT).show();
        }

            remember.setOnCheckedChangeListener((compoundButton, b) -> {
                if (compoundButton.isChecked()) {

                    SharedPreferences preferences = getSharedPreferences("checkbox", MODE_PRIVATE);
                    SharedPreferences.Editor editor = preferences.edit();
                    editor.putString("remember", "true");
                    editor.apply();
                } else if (!compoundButton.isChecked()) {
                    SharedPreferences preferences = getSharedPreferences("checkbox", MODE_PRIVATE);
                    SharedPreferences.Editor editor = preferences.edit();
                    editor.putString("remember", "false");
                    editor.apply();
                }
            });

        btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent intent = new Intent(MainActivity.this, TikTokActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: CodingTask/Communication between activity/app/src/main/java/com/application/communicationbetweenactivity/ui/MainActivity.kt
package com.application.communicationbetweenactivity.ui

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.application.communicationbetweenactivity.R
import com.application.communicationbetweenactivity.interfaceActivity.InterfaceAct
import com.application.communicationbetweenactivity.viewModelActivity.ViewModelAct

import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        ViewModelActivity.setOnClickListener {
            val intent = Intent(this,ViewModelAct::class.java)
            startActivity(intent)
        }

        InterfaceActivity.setOnClickListener {
            val intent = Intent(this,InterfaceAct::class.java)
            startActivity(intent)
        }
    }
}